src/marks/auto.d.ts

```diff
@@ -3,26 +3,74 @@ import type {CompoundMark, Data} from "../mark.js";
 import type {Reducer} from "../reducer.js";
 import type {BinOptions} from "../transforms/bin.js";
 
+// TODO: should jsdoc go here or inside interface?
+type MarkType = "dot" | "line" | "area" | "rule" | "bar";
+
 export interface AutoOptions {
+  /**
+   * The horizontal spatial encoding. If *y* is set without *x*, the *x* channel
+   * will default to the **count** reducer to produce a histogram; you can
+   * override this by passing {reduce: null}.
+   */
   x?: ChannelValue | Reducer | ({value?: ChannelValue; reduce?: Reducer | null; zero?: boolean} & BinOptions);
+  /**
+   * The vertical spatial encoding. If *x* is set without *y*, the *y* channel
+   * will default to the **count** reducer to produce a histogram; you can
+   * override this by passing {reduce: null}.
+   */
   y?: ChannelValue | Reducer | ({value?: ChannelValue; reduce?: Reducer | null; zero?: boolean} & BinOptions);
+  /**
+   * The color encoding, which corresponds to the stroke channel for dots,
+   * lines, and rules, and the fill channel for areas and bars.
+   */
   color?: ChannelValue | Reducer | {value?: ChannelValue; reduce?: Reducer | null; color?: string};
+  /**
+   * The size encoding corresponds to the radius channel of dots; it may
+   * correspond to the length of vectors in the future.
+   */
   size?: ChannelValue | Reducer | {value?: ChannelValue; reduce?: Reducer | null};
+  /**
+   * The horizontal facet dimension.
+   */
   fx?: ChannelValue | {value?: ChannelValue};
+  /**
+   * The vertical facet dimension.
+   */
   fy?: ChannelValue | {value?: ChannelValue};
-  mark?: "dot" | "line" | "area" | "rule" | "bar";
+  /**
+   * A type of mark; for example, the type **bar** encompasses bar, barX, and
+   * barY. Used to guide what sort of mark *auto* should use. It should be thought
+   * of as as an override; *auto* should usually do the right thing without it,
+   * and will try its best with it, but setting the mark type may lead to a
+   * nonsensical plot.
+   */
+  mark?: MarkType;
 }
 
+// TODO: repeat everything here, or is there a better way?
 export interface AutoSpec extends AutoOptions {
   x: {value: ChannelValue; reduce: Reducer | null; zero?: boolean} & BinOptions;
   y: {value: ChannelValue; reduce: Reducer | null; zero?: boolean} & BinOptions;
   color: {value: ChannelValue; reduce: Reducer | null; color?: string};
   size: {value: ChannelValue; reduce: Reducer | null};
   fx: ChannelValue;
   fy: ChannelValue;
-  mark: "dot" | "line" | "area" | "rule" | "bar";
+  mark: MarkType;
 }
 
+/**
+ * Returns a fully-specified *auto* options object, with nothing left undefined.
+ * This is mostly for internal use, but it can be used to “lock down” the
+ * specification of an *auto* mark, or to inspect the decisions its heuristic is
```

Comment:
```suggestion
 * specification of an **auto** mark, or to inspect the decisions its heuristic is
```

Reply:
I’ve been using bold to refer to named options, and italics to refer to parameters (and in some cases scale names… though maybe those should be bold). I still haven’t totally decided on what convention works best, though, so I imagine we’ll eventually take another pass to unify our typographic conventions.